Gnome 49.4 bump (gjs, gnome-control-center, gnome-core-libs, gnome-settings-daemon, gnome-shell, gsettings-desktop-schemas, mutter)

swagtoy opened the pull request:
The biggest bump. Many dependencies, like libadwaita need to be checked in, probably some others, I've lost track.

Gnome-session and GDM are going to probably be split into separate PR's considering the circumstances. They are needed to even build all of this so i've marked this as a draft until we come up with a conclusion on what we should do about this. However my patches at https://github.com/swagtoy/gnome-session-openrc are primarily ready besides a few minor bugs.

~~Some bits are merged in from @e-luks bump except for a gnome-shell patch he grabbed, i'll add it back soon.~~

I'll be a little busy so my responses may be slow on this.

---

Blockers before merge (i think):

- [x] #44092
- [ ] #46097 (to be merged alongside)
- [x] :)

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

swagtoy committed:
dev-libs/gjs: add 1.88.0

Newer versions of gnome shell need gjs >=1.86.0 to compile. This in
turns needs the latest spidermonkey with a patch, but this has since
been applied.

Signed-off-by: Swagtoy <me@ow.swag.toys>
Signed-off-by: Lukas Schmelting <lschmelting@posteo.com>
Co-authored-by: Lukas Schmelting <lschmelting@posteo.com>

swagtoy committed:
gnome-base/gsettings-desktop-schemas: add 49.1

General schemas for new gconf settings. Gnome shell will not run without
this bump.

Also reapplies the old patches. It's still undecided what to do about
the adwaita-fonts dependency for this, so it's being left as is.

Signed-off-by: Swagtoy <me@ow.swag.toys>

swagtoy committed:
gnome-base/gnome-settings-daemon: add 49.1

gnome-session and gnome-control-center requires newer version of g-s-d,
as it starts some new services.
Adds xwayland use flag since g-s-d now has some integration for that.

Signed-off-by: Swagtoy <me@ow.swag.toys>

swagtoy commented:
@e-luks Should `screencast` require devkit?

swagtoy commented:
@pacho2 I probably should've tagged you on this long ago. Are you interested in taking a little skim?

swagtoy commented:
As for bumping GDM 49, we may _not_ need to create new group/acct ebuilds: see https://github.com/elogind/elogind/issues/323#issuecomment-3574375675.

Frankly, once elogind bumps, that should make GDM usable for launching other environments. ~~Question remains on if putting a blocker if it can't launch gnome-shell is worth it.~~ Oh right, it still _does_ launch gnome-session, I forgot for a hot sec. Still, this is good news, so sharing here.

swagtoy commented:
Should I drop gnome-49.0/49.1 ebuilds when pulling 49.2?

zyxhere commented:
> Should I drop gnome-49.0/49.1 ebuilds when pulling 49.2?

Yes

jpnssl commented on gnome-base/gnome-settings-daemon/gnome-settings-daemon-49.0.ebuild:
```diff
@@ -0,0 +1,146 @@
+		$(meson_use smartcard)
+# TODO
+#		$(meson_use input_devices_wacom wacom)
+#		$(meson_use wayland)
```
```suggestion
		$(meson_use X x11)
		-Dxwayland=true
```

jpnssl commented on gnome-base/gnome-settings-daemon/gnome-settings-daemon-49.0.ebuild:
```diff
@@ -0,0 +1,146 @@
+	|| (
+		media-libs/libcanberra-gtk3
+		>=media-libs/libcanberra-0.25[gtk3(-)]
+	)
```
```suggestion
	>=media-libs/libcanberra-0.25
```
upstream dropped libcanberra-gtk3 in favor of plain libcanberra

jpnssl commented on gnome-base/gnome-settings-daemon/gnome-settings-daemon-49.0.ebuild:
```diff
@@ -0,0 +1,146 @@
+		>=dev-libs/libwacom-0.7:=
+		>=x11-libs/pango-1.20.0
+		x11-libs/gdk-pixbuf:2
+	)
```
```suggestion
```
upstream removed the wacom plugin

swagtoy replied:
Oh, did they? Is it all libinput now?

jpnssl commented on gnome-base/gsettings-desktop-schemas/gsettings-desktop-schemas-49.0.ebuild:
```diff
@@ -0,0 +1,38 @@
+IUSE="+introspection"
+
+BDEPEND="
+	introspection? ( >=dev-libs/gobject-introspection-1.54:= )
```
```suggestion
	introspection? ( >=dev-libs/gobject-introspection-1.82.0-r2:= )
```
there recently was a general version bump for all gobject-introspection dependencies ([link](https://github.com/gentoo/gentoo/commit/d83e25b6a59578c519b2910640b5cb4205f0275c))

swagtoy replied:
I am actually aware of this one and have noticed it before; was in some other things as well :) Thanks!

zyxhere commented on gnome-base/gnome-settings-daemon/gnome-settings-daemon-49.0.ebuild:
```diff
@@ -0,0 +1,146 @@
+LICENSE="GPL-2+ LGPL-2+"
+SLOT="0"
+
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~loong ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux"
```
Gentoo prefix I'm pretty sure

swagtoy commented:
ok... bumped and built OK. testing runtime here, but still need to apply stuff. sleepy and got a long drive tomorrow so no more... need rest, but heres something new so i dont forget.. gnome-shell

```
 * QA Notice: The ebuild is installing to one or more unexpected directories:
 * 
 *   /usr/share/doc/shell
 *   /usr/share/doc/st
 * 
 * Please fix the ebuild to use correct FHS/Gentoo policy paths.
 * 
 * QA Notice: This package seems to install gtk-doc documentation into the
 * following location(s):
 * 
 *   /usr/share/doc/shell
 *   /usr/share/doc/st
 * 
 * gtk-doc documentation must always be installed into /usr/share/gtk-doc.
 * For more details, please see the GNOME team policy page:
 * https://wiki.gentoo.org/wiki/Project:GNOME/Gnome_Team_Ebuild_Policies#Handling_gtk-doc_documentation
```

e-luks commented on gnome-base/gnome-settings-daemon/gnome-settings-daemon-49.1.ebuild:
~`	x11-libs/libXi`~
https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/commit/37102a6056c37ec497dbb096770b0c43762865e2#0cc1139e3347f573ae1feee5b73dbc8a8a21fcfa

swagtoy commented on x11-wm/mutter/mutter-49.0.ebuild:
```diff
@@ -0,0 +1,272 @@
+# native backend requires gles3 for hybrid graphics blitting support, udev and a logind provider
+REQUIRED_USE="
+	|| ( X wayland )
+	devkit? ( screencast )
```
> Devkit also requires wayland? https://gitlab.gnome.org/GNOME/mutter/-/blob/49.2/meson.build?ref_type=tags#L357

Where did you infer that? Don't see it.

swagtoy commented:
Unmarked as draft now, since I've now taken the care to look through most bits and fixed/rebased all the broken stuff (i think...); most reviews seemed correct though there are a few deps i want to double check git log meson.build etc against.

Doesn't mean this can get merged; there is the glycin blocker and the gdm (should be easy tbh)/gnome-session{,-openrc} bits will get done soon here in PRs, mostly used as an indicator that I am a bit more confident with the stuff here than before when i was mostly focused on the openrc stuff...

jpnssl commented on x11-wm/mutter/mutter-49.0.ebuild:
```suggestion
IUSE="bash-completion debug devkit elogind gnome gtk-doc input_devices_wacom +introspection screencast sysprof systemd test udev +wayland X +xwayland video_cards_nvidia"
# native backend requires gles3 for hybrid graphics blitting support, udev and a logind provider
REQUIRED_USE="
	|| ( X wayland )
	devkit? ( screencast )
	gtk-doc? ( introspection )
	wayland? ( ^^ ( elogind systemd ) udev )
	test? ( screencast wayland )
	xwayland? (wayland)"
```
If xwayland is enabled while wayland is disabled, the build system will throw an error. IMHO, enabling wayland by default makes sense as mutter is very much wayland-first now.

swagtoy committed:
acct-user/gdm-greeter: new package, add 0

Signed-off-by: Swagtoy <me@ow.swag.toys>

swagtoy committed:
gnome-base/gdm: add 49.2

Signed-off-by: Swagtoy <me@ow.swag.toys>

swagtoy committed:
gnome-base/gnome-session: add 49.2

Drops most of the x11 dependencies since they have been fixed with gtk4.

Also introduces a new dependency on gnome-session-openrc. This is
required as gnome-session adds harder dependencies on gnome-session
overall, so a new runtime leader/ctl drop-in was created for openrc
systems.

In addition to that, the old elogind patches were re-adapted to also
allow for this new leader to work: this is what makes
gnome-session-openrc an RDEPEND, so it fills that role with its own
specialized leader/ctl tools as well as some init scripts.

Older patches were dropped as they have been resolved upstream and via
the new systemd leader.

Signed-off-by: Swagtoy <me@ow.swag.toys>

swagtoy committed:
gnome-base/gnome-session-openrc: new package, add 49.0

See the parent commit for gnome-session which explains a bit: this is
the runtime leader/ctl tool for gnome-session specialized for openrc
systems.

Signed-off-by: Swagtoy <me@ow.swag.toys>

lkraav commented:
Is this still blocked by something?

swagtoy commented:
Yes; by time, our own priorities/lives, and the gdm PR which still needs to get reviewed. 

If it makes you feel better; I've already begun work on GNOME 50, in fact I'm using it right now but the ebuilds need some polish since X was dropped. Only on a local branch for now. I want to wait until this PR is in before I publish that so I can rebase against any suggestions.

If you want it out faster or could give a hand, reviewing and personally testing it yourself is a _huge_ help. Everything is ready, but so far I haven't caught onto anyone testing it atm :/

I do need to apply a suggestion in here, I wanted to do it last night, but I will do it today shortly here

swagtoy commented:
Also, the last blocker was Glycin, which only recently got merged in

lkraav commented:
> If you want it out faster or could give a hand, reviewing and personally testing it yourself is a _huge_ help. Everything is ready, but so far I haven't caught onto anyone testing it atm :/

If it's mostly ready to daily-drive, I can indeed jump in. Was running https://github.com/flint2/gentoo/tree/gnome48 for months before it hit main tree.

X-less G50 definitely sounds interesting!

swagtoy commented:
Well, ive been daily driving it since I opened it and its been good :) but really, if you want it out faster, then please test it. If at worst it breaks, there isn't really anything stopping you from rolling back anyway.

swagtoy commented:
@jpnssl I can't reply because Github is buggy as all hell, but anyway, yes, after looking I agree that wayland should be a required use for devkit, so ticking it on.....

swagtoy committed:
gnome-extra/gnome-shell-extensions: add 49.0

Dropped gtk3 dependency in place of gtk4

Signed-off-by: Swagtoy <me@ow.swag.toys>

lkraav commented:
> Well, ive been daily driving it since I opened it and its been good :) but really, if you want it out faster, then please test it. If at worst it breaks, there isn't really anything stopping you from rolling back anyway.

Is there any real-time chat channel you hang out at? I don't see this nick at IRC `#gentoo-desktop`.

zyxhere commented:
On Wed, 2026-04-22 at 01:20 -0700, Leho Kraav wrote:
> lkraav left a comment (gentoo/gentoo#44112)
> > Well, ive been daily driving it since I opened it and its been good
> > :) but really, if you want it out faster, then please test it. If
> > at worst it breaks, there isn't really anything stopping you from
> > rolling back anyway.
> Is there any real-time chat channel you hang out at? I don't see this
> nick at IRC #gentoo-desktop.

gentoo-gnome

pacho2 commented on dev-libs/gjs/gjs-1.88.0.ebuild:
gtk4 is in a different category: gui-libs/gtk

pacho2 commented:
Hello,

I have issues applying the PR with the current tree. It seems that mutter and gnome-shell commits would need to be rebased.

dev-libs/gjs: I am unsure if maybe adding it to the current package.mask glib entry and, when everything is pushed, unmask all together
```
# Lukas Schmelting <lschmelting@posteo.com> (2025-10-03)
# Breaks e.g. in combination with gnome-shell-48.4, gettext/gjs (bug #963815)
>=dev-libs/glib-2.86.0
>=dev-libs/gobject-introspection-1.86.0
>=dev-libs/gobject-introspection-common-1.86.0
>=dev-util/gdbus-codegen-2.86.0
>=dev-util/glib-utils-2.86.0
```

pacho2 commented on dev-libs/gjs/gjs-1.88.0.ebuild:
From meson files, it seems that any spidermonkey:140 should work, if that is the case, we don't need to specify the minor versions... if not, it is fine as it is now

swagtoy replied:
Yeah that's a good idea. There was originally an issue with Spidermonkey & GJS originally that I had to send a patch a long time ago for. I think that's what I accounted for initially (i wrote this ebuild quite a good while ago..). juippis ended up adding it (0007-bmo-1973994-mozjs-140.pc-does-not-contain-DXP_UNIX_-o.patch from https://dev.gentoo.org/~juippis/mozilla/patchsets/spidermonkey-140-patches-02.tar.xz) since then.

But a glimpse at the patch says it was initially created for 140.1.0, so I think I will concur and remove the version: if this breaks for some weird reason, we can always undo it later.

pacho2 commented on gnome-base/gsettings-desktop-schemas/gsettings-desktop-schemas-49.1.ebuild:
```diff
@@ -0,0 +1,43 @@
+
+LICENSE="LGPL-2.1+"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~loong ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~x64-macos"
```
I get this error:
  UnknownKeywords: version 49.1: unknown KEYWORDS: '~amd64-linux', '~x86-linux'

You simply need to drop those keywords

pacho2 reviewed (commented):
I am not sure about the following commit message:
```
gnome-base/gnome-settings-daemon: add 49.1
gnome-session requires that these are bumped.
```

Could it be more clear? (or dropped, if not needed anymore). Thanks

pacho2 commented on gnome-base/gnome-settings-daemon/gnome-settings-daemon-49.1.ebuild:
```diff
@@ -0,0 +1,138 @@
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{10..14} )
```
It could probably be updated to 11..14 at this time

swagtoy replied:
Yeah, i remember talking with eluks about this, but i'll change it

pacho2 commented on gnome-base/gnome-settings-daemon/gnome-settings-daemon-49.1.ebuild:
```diff
@@ -0,0 +1,138 @@
+
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~loong ~ppc ~ppc64 ~riscv ~sparc ~x86"
+
+IUSE="+colord +cups debug elogind modemmanager networkmanager smartcard systemd test +xwayland X"
```
metadata.xml will need a description for xwayland USE flag

pacho2 commented on dev-libs/gjs/gjs-1.88.0.ebuild:
Copyright years need to be updated

swagtoy replied:
very weird... it was already up to date for me :?

pacho2 commented on gnome-base/gnome-core-libs/gnome-core-libs-49.0.ebuild:
```diff
@@ -0,0 +1,41 @@
+IUSE="cups python"
+
+# when unmasking for an arch
+# double check none of the deps are still masked !
```
I don't think that these comments are needed. pkgcheck will show the errors when trying to add some arch with unkeyworded deps

swagtoy replied:
I did not write those comments

pacho2 commented on gnome-base/gnome-core-libs/gnome-core-libs-49.0.ebuild:
```diff
@@ -0,0 +1,41 @@
+
+# Note to developers:
+# This is a wrapper for the core libraries used by GNOME
+RDEPEND="
```
Lots of dep versions were downgraded... I wonder if it is not based on an older ebuild Anyway, if you wish, you can simply drop the metas update and postpone it for later

swagtoy replied:
Yeah, lets drop it. To be honest, I haven't actually looked at this ebuild much at all. I planned to just drop it entirely before because there is uhh... nothing really changed anyway :)

There was some debate regarding if we should put glycin in it and I wasn't exactly sure. Maybe the decision can become easier as its usage is a lot more widespread.

swagtoy replied:
Also i completely forgot to mention; yes, it was based off of an older ebuild :P sorry!

pacho2 commented:
There is a problem with the gjs commit message:
```
commit
  BadCommitSummary: commit d3290cb57d5a, summary missing package version '1.88.0': 'dev-libs/gjs: add 1.87.2'
```

swagtoy commented:
Yeah, i bumped it last night but forgot to rename it, sorry! ;')

swagtoy commented:
I just need to rebase it now... will come back to that...

pacho2 reviewed (commented):
I get this warning with pkgcheck scan --commits:
InvalidCommitMessage: commit 648ddc2b704b: line 2 greater than 80 chars: 'Soft depends on glycin-2.0 (#44092) and glycin loaders for the wallpapers, but not required.'

pacho2 commented on dev-libs/gjs/gjs-1.88.0.ebuild:
You need to drop ">=", that explains the following "pkgcheck scan --commits" error:
```  InvalidDepend: version 1.88.0: failed parsing DEPEND: '>=dev-libs/glib-2.86.0:2 dev-libs/libffi:= >=dev-libs/gobject-introspection-1.86.0:= >=dev-lang/spidermonkey:140 x11-libs/cairo[X,glib] readline? ( sys-libs/readline:0= ) sysprof? ( >=dev-util/sysprof-capture-3.40.1:4 ) test? ( sys-apps/dbus >=x11-libs/gtk+-3.20:3[introspection] >=gui-libs/gtk-4[introspection] )' is unparseable: flagged token- invalid package atom: '>=dev-lang/spidermonkey:140'
  InvalidRdepend: version 1.88.0: failed parsing RDEPEND: '>=dev-libs/glib-2.86.0:2 dev-libs/libffi:= >=dev-libs/gobject-introspection-1.86.0:= >=dev-lang/spidermonkey:140 x11-libs/cairo[X,glib] readline? ( sys-libs/readline:0= )' is unparseable: flagged token- invalid package atom: '>=dev-lang/spidermonkey:140'
```

pacho2 commented on gnome-base/gnome-shell/gnome-shell-49.4.ebuild:
```diff
@@ -0,0 +1,195 @@
+
+	app-arch/gnome-autoar
+	dev-libs/json-glib
+	net-libs/libsoup
```
It needs :3.0 slot

pacho2 commented on gnome-base/gnome-shell/gnome-shell-49.4.ebuild:
```diff
@@ -0,0 +1,195 @@
+	gnome-base/librsvg:2[introspection]
+	gui-libs/libadwaita:1[introspection]
+
+	>=gnome-base/gnome-session-2.91.91
```
This was downgraded from 48.0-r2, is it expected? (I couldn't find the change)

swagtoy replied:
It's not, you can see this in the current gnome-shell-48.5.ebuild. I based this PR initially from 48.4 or something, and I haven't tracked changes since.

To be really honest, I haven't gotten to rebasing this entire PR yet. Currently, I am doing that right now. I apologize that there are things that have since changed that you've been reviewing :(

swagtoy replied:
I've went ahead and switched this to ver_cut, actually.

EDIT: Not anymore, after a discussion over IRC.

pacho2 commented on gnome-base/gnome-shell/gnome-shell-49.4.ebuild:
```diff
@@ -0,0 +1,195 @@
+BDEPEND="
+	>=dev-build/meson-1.3.0
+	dev-libs/libxslt
+	>=dev-util/gdbus-codegen-2.45.3
```
This was downgraded from previous 2.80.5-r1, was it done on purpose?

swagtoy replied:
No, now rebased.

pacho2 commented on gnome-base/gnome-shell/gnome-shell-49.4.ebuild:
```diff
@@ -0,0 +1,195 @@
+	virtual/pkgconfig
+	test? (
+		sys-apps/dbus
+		=x11-wm/mutter-49.0[test]
```
I would revert to x11-wm/mutter[test] as the right version is already pulled before and, here, we only pretend to ensure that mutter is built with tests enabled.

pacho2 commented on gnome-base/gnome-shell/gnome-shell-49.6.ebuild:
```diff
@@ -0,0 +1,195 @@
+# avoid circular dependency, see bug #546134
+# keep in sync with gnome-shell-${PV} bug 961591
+PDEPEND="
+	>=gnome-base/gdm-49[introspection(+)]
```
Maybe "49" could be replaced by "$(ver_cut 1)" to not need to remember to update it on every major bump. Also, that makes me wonder if maybe adding here also your gdm ebuilds from the other PR as, anyway, we will need to push them together

swagtoy replied:
Yeah. I think this was like that before... there are few instances i recall from ebuilds where the version number here is hardcoded for some reason, like -49* was another one I think I saw...

swagtoy replied:
@pacho2 tbh i split gdm and gnome-session et al up into a separate PR just to make it easier to review, esp since there was a new ebuild there. They do indeed need to be pushed together, as mentioned in the body.

Do you want me to cherry-pick those here as well for convenience and I can close the other?

swagtoy replied:
OK, so i had a discussion with Leio about this over IRC. He seems to be against  the change, over the potentiality of one of the majors version not matching over time. Though this contradicts the comment above, I think I'm going to leave this alone for now.

I _do_ agree personally we should do this, but I'd wait for a more resolution conclusion on his part.

pacho2 replied:
OK; then, either keep it as it was in the previous version or, if there was an exact bug/issue needing >= 49 version, you can keep it. But, in that case, as that would be a puntual version bump and not something that we pretend to change for every major version bump, I wouldn't mention that bug report (as it would be misleading)

pacho2 commented on gnome-base/gnome-control-center/gnome-control-center-49.7.ebuild:
```diff
@@ -0,0 +1,227 @@
+	)
+	>=media-libs/libpulse-2.0[glib]
+	>=gui-libs/gtk-4.17.1:4[X,wayland=]
+	>=gui-libs/libadwaita-1.8_alpha:1
```
Because of this change, >=gnome-base/gnome-control-center-49 will need to be added to the rust mask at features/wd40/package.mask

pacho2 commented on gnome-base/gnome-control-center/gnome-control-center-49.5.ebuild:
```diff
@@ -0,0 +1,227 @@
+	dev-libs/libxml2:2=
+	>=sys-power/upower-1.90.6:=
+	>=dev-libs/libgudev-232
+	>=x11-libs/libX11-1.8
```
It seems that libX11 and libXi requirements were moved into an optional x11 support:
```
-x11_dep = dependency('x11', version: '>= 1.8')
-xi_dep = dependency('xi', version: '>= 1.2')
 epoxy_dep = dependency('epoxy')
 gcr_dep = dependency('gcr-4', version: '>= 4.1.0')
 pwquality_dep = dependency('pwquality', version: '>= 1.2.2')
@@ -255,6 +266,14 @@ config_h.set10('HAVE_CUPS_HTTPCONNECT2',
                cc.has_function('httpConnect2', dependencies: cups_dep),
                description: 'Define if httpConnect2() is available in CUPS')
 
+# X11
+have_x11_support = get_option('x11')
+if have_x11_support
+  x11_dep = dependency('x11', version: '>= 1.8')
+  xi_dep = dependency('xi', version: '>= 1.2')
+endif
+config_h.set('HAVE_X11', have_x11_support, description: 'Defined if X11 support is enabled')
```

swagtoy replied:
Went ahead and added an X use flag

pacho2 commented on gnome-base/gnome-control-center/gnome-control-center-49.7.ebuild:
```diff
@@ -0,0 +1,227 @@
+	)
+	ibus? ( >=app-i18n/ibus-1.5.2 )
+	networkmanager? (
+		>=net-libs/libnma-1.10.2
```
This should probably be raised:
```
   # NetworkManager
   network_manager_deps = [
-    dependency('libnm', version: '>= 1.24.0'),
+    dependency('libnm', version: '>= 1.52.0'),
     dependency('libnma-gtk4', version: '>= 1.10.2'),
     mm_dep,
   ]
```

swagtoy replied:
Are you certain? That's libnma, which is version 1.10.2. networkmanager is already at 1.52.0

pacho2 replied:
I think that you are right :)

pacho2 commented on gnome-base/gnome-control-center/gnome-control-center-49.5.ebuild:
```diff
@@ -0,0 +1,227 @@
+SLOT="2"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc ~ppc64 ~riscv ~x86"
+
+IUSE="+bluetooth +cups debug elogind +gnome-online-accounts +ibus input_devices_wacom kerberos +geolocation networkmanager systemd test wayland"
```
It seems that x11/wayland handling changed:
```
--- /tmp/mgorny-dev-scripts/portage/gnome-base/gnome-control-center-48.7/work/gnome-control-center-48.7/meson_options.txt       2026-03-13 12:59:22.000000000 +0100
+++ /tmp/mgorny-dev-scripts/portage/gnome-base/gnome-control-center-49.5/work/gnome-control-center-49.5/meson_options.txt       2026-02-27 13:49:15.000000000 +0100
@@ -5,7 +5,7 @@ option('ibus', type: 'boolean', value: t
 option('privileged_group', type: 'string', value: 'wheel', description: 'name of group that has elevated permissions')
 option('snap', type: 'boolean', value: true, description: 'build with Snap support')
 option('tests', type: 'boolean', value: true, description: 'build tests')
-option('wayland', type: 'boolean', value: true, description: 'build with Wayland support')
+option('x11', type: 'boolean', value: false, description: 'build with X11 support')
 option('profile', type: 'combo', choices: ['default','development'], value: 'default')
 option('malcontent', type: 'boolean', value: false, description: 'build with malcontent support')
 option('distributor_logo', type: 'string', description: 'absolute path to distributor logo for the About panel')
```

pacho2 commented on gnome-base/gnome-control-center/gnome-control-center-49.5.ebuild:
```diff
@@ -0,0 +1,227 @@
+src_configure() {
+	# -Werror=strict-aliasing
+	# https://bugs.gentoo.org/889008
+	# https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/2563
```
These two bugs should be fixed in this version

pacho2 commented on gnome-base/gnome-control-center/gnome-control-center-49.5.ebuild:
```diff
@@ -0,0 +1,227 @@
+DESCRIPTION="GNOME's main interface to configure various aspects of the desktop"
+HOMEPAGE="https://apps.gnome.org/Settings"
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+# FIXME: pacho must upload the patchos to the ~pacho patchset
```
I just uploaded it. You should be able to download it when using:
SRC_URI+=" https://dev.gentoo.org/~pacho/${PN}/${PN}-49.5-patchset.tar.xz"

swagtoy replied:
Thanks!

pacho2 commented on x11-wm/mutter/mutter-49.4.ebuild:
```diff
@@ -0,0 +1,272 @@
+	introspection? ( >=dev-libs/gobject-introspection-1.82.0-r2:= )
+	test? (
+		>=x11-libs/gtk+-3.19.8:3[X,introspection?]
+		>=dev-utils/umockdev-0.3.0
```
there is a typo -> dev-util/ (you need to drop the "s")

swagtoy replied:
A hard lesson I _really_ should've learned earlier was that I should start actually running tests for things I am dealing with ;-)

Sorry about this!

pacho2 commented on x11-wm/mutter/mutter-49.4.ebuild:
```diff
@@ -0,0 +1,272 @@
+	SRC_URI=""
+	SLOT="0/17" # This can get easily out of date, but better than 9967
+else
+	KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
```
You will need to drop all but amd64 until other arches start working on https://bugs.gentoo.org/973153

Thanks

swagtoy committed:
gnome-base/gnome-shell: add 49.6

Soft depends on glycin-2.0 (#44092) and glycin loaders for the wallpapers, but not required.

Bug: https://bugs.gentoo.org/961591
Signed-off-by: Swagtoy <me@ow.swag.toys>
Signed-off-by: Lukas Schmelting <lschmelting@posteo.com>
Co-authored-by: Lukas Schmelting <lschmelting@posteo.com>

swagtoy committed:
x11-wm/mutter: add 49.5

Depends on glycin-2.0 (#44092). There are various upstream patches that
should be applied here but since 49.1 is releasing today (or tomorrow
(or the day after that)) I am going to exclude them for now so we don't
have to juggle patches (unless someone desparately needs 49.0 and not 49.1).

A new dependency on devkit is also introduced for screensharing and nesting.

Signed-off-by: Swagtoy <me@ow.swag.toys>
Signed-off-by: Lukas Schmelting <lschmelting@posteo.com>
Co-authored-by: Lukas Schmelting <lschmelting@posteo.com>

swagtoy committed:
gnome-base/gnome-control-center: add 49.7

This includes a rebase of the old patches, a blueprint dep, and some dep
version bumps from upstream.

Signed-off-by: Swagtoy <me@ow.swag.toys>

jpnssl commented on gnome-base/gnome-shell/gnome-shell-49.4.ebuild:
```diff
@@ -0,0 +1,195 @@
+	>=media-libs/libpulse-2[glib]
+	dev-libs/libical:=
+
+	gui-libs/gtk:4[introspection]
```
```suggestion
```
We already have a stronger gtk dependency check on line 28, so this one is not needed.

jpnssl commented on gnome-base/gnome-shell/gnome-shell-49.6.ebuild:
```diff
@@ -0,0 +1,195 @@
+	net-libs/libsoup:3.0[introspection]
+	>=sys-power/upower-0.99:=[introspection]
+	gnome-base/librsvg:2[introspection]
+	gui-libs/libadwaita:1[introspection]
```
```suggestion
	gui-libs/libadwaita:1[introspection]
	dev-libs/glib:2[introspection]
```
At runtime, we need glib's GIRs.

pacho2 replied:
Are you sure that they are neither needed at build time? I was wondering about simply adding [introspection] to the glib DEPEND line before 🤔

jpnssl replied:
After taking a closer look, turns out you are correct. They are in fact needed at build time. Sneaky GIR includes hiding in a subdirectory!

jpnssl commented on gnome-base/gnome-shell/gnome-shell-49.6.ebuild:
```diff
@@ -0,0 +1,195 @@
+	>=dev-libs/gobject-introspection-1.86.0:=
+	>=dev-libs/gjs-1.85.90[cairo(+)]
+	>=gui-libs/gtk-4:4[X?,introspection,wayland?]
+	>=x11-wm/mutter-49.0:0/17[introspection,test?]
```
gnome-shell's x11 configuration mirrors that of mutter. To determine what to enable at build time (and what to link against), the meson script looks at the installed build of mutter and enables the corresponding features in gnome-shell. This means that if gnome shell is built against a version of mutter with USE=X or USE=xwayland set, and a user decides to rebuild mutter with any of these USE-flags disabled, gnome-shell will probably crash.

The only solution I can think of to make sure this cannot happen is to add a corresponding USE=xwayland to gnome-shell and to lock it and USE=X to mutter:
```suggestion
	>=x11-wm/mutter-49.0:0/17[X=,introspection,test?,xwayland=]
```

pacho2 replied:
Hello.

I am a bit unsure about these changes at this stage because xwayland option is not honored by gnome-shell meson files and, until now, we did not see issues with this. For this PR I would then skip this change (and keep the logic used for 48.x versions), and maybe another PR could be opened later to see how to cover this exact issue.

Thanks

pacho2 replied:
I was thinking on xwayland USE... for "X", I agree that it should probably be modified to "X="

jpnssl replied:
Good point, better not bring too much bikeshedding into this.

swagtoy replied:
I'll leave this alone.

This will probably all need to be rewired for GNOME 50 anyway (which i have stashed right now... will rebase it on these recent changes later :)

jpnssl commented on gnome-base/gnome-shell/gnome-shell-49.6.ebuild:
```diff
@@ -0,0 +1,195 @@
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
+
+IUSE="X elogind gtk-doc +ibus +networkmanager pipewire systemd test wayland"
```
```suggestion
IUSE="X elogind gtk-doc +ibus +networkmanager pipewire systemd test wayland +xwayland"
```

pacho2 replied:
(the same doubts exposed in previous comments)

swagtoy commented:
(fwiw, i havent pushed my changes yet even if i marked stuff as resolved. i still want to test bits and change some commit messages. should have it in by tomorrow afternoon at the soonest)

pacho2 commented:
> (fwiw, i havent pushed my changes yet even if i marked stuff as resolved. i still want to test bits and change some commit messages. should have it in by tomorrow afternoon at the soonest)

Just on time as I was already reviewing the first commits of the rebased ones. Regarding gdm PR... do as you wish: if you merge them here, fine; if you keep the splitted one, I will try to push both at the same time.

swagtoy commented:
No worries. I'll push those here and close the other tomorrow, sorry for doing it like that :P thought it'd clean it up...

swagtoy committed:
profiles/features/wd40: gnome-base/gnome-control-center needs rust

Due to added libadwaita dep

Signed-off-by: Swagtoy <me@ow.swag.toys>

reagentoo commented:
Thank you. But gdm is dead..

thesamesam commented:
Please file a bug. That comment isn't actionable.
